src/super_gradients/recipes/training_hyperparams/coco2017_yolox_train_params.yaml

```diff
@@ -0,0 +1,35 @@
+defaults:
+  - default_train_params
+
+max_epochs: 300
+lr_mode: cosine
+cosine_final_lr_ratio: 0.05
+lr_warmup_epochs: 5
+warmup_bias_lr: 0.0
+warmup_momentum: 0.9
+initial_lr: 0.01
+warmup_mode: yolov5_warmup
+
+loss: yolox_loss
+criterion_params: ??? # will be overridden in train conf as it includes sg_model
+loss_logging_items_names: [iou, obj, cls, l1, num_fg, Loss]
+
+optimizer: SGD
+optimizer_params:
+  momentum: 0.9
+  weight_decay: 0.0005  # will be scaled according to effective batch size
+  nesterov: True
+
+ema: True
+mixed_precision: False
+
+valid_metrics_list:
+  - _target_: super_gradients.training.metrics.DetectionMetrics
```

Comment:
Using a factory since 4511cd8
Would be nice, not sure how. arch_params isn't visible in this yaml yet, and I can't leave `num_cls: ???` because hydra attempts to construct DetectionMetrics before it reaches overriding in the main yaml. Need help on this.

Reply:
maybe if we change the orders of the defaults in coco2017_yolox.yaml (lines 2,4) then the these two will be loaded in a different order